sync-core/src/main/java/com/cloudant/sync/datastore/BasicDatastore.java

```diff
@@ -1968,7 +2034,7 @@ public BasicDocumentRevision call(SQLDatabase db) throws Exception {
         } catch (InterruptedException e) {
             logger.log(Level.SEVERE,"Failed to delete document",e);
         } catch (ExecutionException e) {
-            logger.log(Level.SEVERE,"Failed to delete document",e);
+            throw new DocumentException("Failed to delete document",e);
```

Comment:
Worth appending `e`'s message to these generic messages that appear around the place? I guess the stack trace might contain `e`'s message?